htdocs/web_portal/javascript/leafletembed.js

```diff
@@ -1,46 +1,37 @@
-//function to display google map. Adapted by George Ryall from the following tutorials:
-// * https://developers.google.com/maps/articles/phpsqlajax_v3
-// * https://developers.google.com/maps/documentation/javascript/tutorial
-// * http://www.w3schools.com/googleAPI/
-// * The example code here : http://gmaps-samples-v3.googlecode.com/svn/trunk/xmlparsing/ 
+// javascript file to display a leaflet map.
+// Based on https://leafletjs.com/examples/quick-start/ and
+// https://switch2osm.org/using-tiles/getting-started-with-leaflet/
 
-google.maps.event.addDomListener(window, 'load', initialize);
-
-var infowindow;
 var map;
 
-function initialize() {
-    var mapProp = {
-        //Starting position of map
-        center:new google.maps.LatLng(30,0),
-        zoom:2,
-        //Enable zoom control, but move it to the left and make it small
-        zoomControl:true,
-        zoomControlOptions: {
-            style:google.maps.ZoomControlStyle.SMALL,
-            position:google.maps.ControlPosition.RIGHT_TOP
-        },
-        //Allow map type choiuce, but move to bottom left
-        mapTypeControl:true,
-        mapTypeControlOptions: {
-            position:google.maps.ControlPosition.TOP_RIGHT   
-        },
-        //Turn off pan control - it clutters map and users can drag if needed
-        panControl:false,
-        //turn of street view controller
-        streetViewControl:false,
-        //Options:ROADMAP/SATELLITE/HYBRID/TERRAIN   
-        mapTypeId:google.maps.MapTypeId.ROADMAP
-    };
+// Function to display a leaflet map with markers from GOCDB Site data.
+function initmap() {
+	// set up the map
+	map = new L.Map('map');
+
+	// create the tile layer with correct attribution.
+	var osmUrl='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+	var osmAttrib='Map data © <a href="https://openstreetmap.org">OpenStreetMap</a> contributors';
+	var osm = new L.TileLayer(osmUrl, {
+        attribution: osmAttrib,
+        // A min zoom of 2 gives you a view of most of the world.
+        minZoom: 2,
```

Comment:
This seems a bit limiting if you imagine all the markers in place. Why not let people zoom out as far as they want?

Reply:
because (given the current centre of the map), zooming out any further means the map gets smaller than the box it's and you get grey bars at top or bottom.

Reply:
We do allow that at the moment though. You can still have the default zoom level as 2. It would be good to see how it looks with all the markers.

Reply:
> It would be good to see how it looks with all the markers.

[ta dah!](https://host-130-246-214-232.nubes.stfc.ac.uk/portal/)

Reply:
Thanks. Looks good with all the markers. Nicer than the Google one in my opinion.

I still think we should allow the zoom out option and use something close to the previous centre. If anything we could limit the zoom in as many of the markers are not that accurate.

Reply:
I have re-centred the map to something closer to what we already have. I disagree about the zoom out/in options.